tests/runner.py

```diff
@@ -216,8 +216,8 @@ def build(self, src, dirname, filename, output_processor=None, main_file=None, a
                ['-I', dirname, '-I', os.path.join(dirname, 'include')] + \
                map(lambda include: '-I' + include, includes) + \
                ['-c', f, '-o', f + '.o']
-        output = Popen(args, stdout=PIPE, stderr=self.stderr_redirect if not DEBUG else None).communicate()[0]
-        assert os.path.exists(f + '.o'), 'Source compilation error: ' + output
+        output = subprocess.check_call(args, stdout=PIPE, stderr=self.stderr_redirect if not DEBUG else None)
```

Comment:
I removed the `PIPE` - how is that? Seems cleaner to me. The only downside is we might get some clang frontend warnings, but if those are annoying, we can fix those on a per-test basis, tests shouldn't have warnings probably anyhow.